### ISSUE TITLE
Implement text windows

### ISSUE THREAD

**Issue by bobbimanners** [bobbimanners/Applecorn#154]
This is half implemented at present (Sep 15.)

Things to do:
- `CLS` clears whole lines.  Needs to be reworked to respect left and right window edges.
- Scrolling also needs to be reworked to respect left and right window edges.
- Copy cursor is messed up when window is not the full screen.
- Special attention will have to be paid to text in `MODE 1` (Apple `HGR`).  Need a custom clear routine.

**Comment by bobbimanners** [comment]
For clearing regions of the `HGR` screen (`MODE 1`) I can probably use FDraw's `FillRect` call.

**Comment by bobbimanners** [comment]
Fixing `CLS` and scrolling is a matter of making `CLREOL` do the right thing (ie: clear from current cursor column to the right hand side of the text window.)  I have this working for 40 column mode.

**Comment by bobbimanners** [comment]
Implemented `CLREOL2` which works correctly in both 40 and 80 column text mode.  This is currently checked into branch `clreol`.  There are some loose ends:
 - Needs to support the GS, in the same manner as `CLREOL`.
 - We also need to fix this for `HGR` / `MODE 1` graphics mode (ie: `HCLREOL`).
Once this is done we can remove the old `CLREOL` and rename `CLREOL2`.

**Comment by bobbimanners** [comment]
`SCROLLER` still needs some work to only scroll between the left and right edges of the text window.  Probably can do it in a similar way to `CLREOL`.

**Comment by bobbimanners** [comment]
`CLREOL2` has been renamed to `CLREOL` and the old routine deleted.  Added support for the GS using long writes to banks E0 and E1.

`HGR` / `MODE 1` still needs to be addressed.

**Comment by bobbimanners** [comment]
`SCR1LINE` is now working with text windows for both the //e and GS, so scrolling works as it should in the text modes.

The remaining issue is the `MODE 1` / `HGR` mode.

**Comment by bobbimanners** [comment]
Scrolling in `MODE 1` works again, but doesn't respect text windows yet.  Should be simple (famous last words!)

**Comment by bobbimanners** [comment]
`HGRSCR1L` needs to be updated to respect the text windows.

`HCLRLINE` should clear to end of line, and respect the text window.

**Comment by bobbimanners** [comment]
`HCLRLINE` is fixed.

**Comment by bobbimanners** [comment]
`HGRSCR1L` is fixed now so that scrolling of text windows works in `MODE 1` too.

**Comment by bobbimanners** [comment]
Only loose end I am aware of is the copy cursor.  Will close this and open a new ticket for the copy cursor.
